### DISCUSSION TITLE
Ignore/disable old meters

### DISCUSSION THREAD

**Discussion by psfshr** [BottlecapDave/HomeAssistant-OctopusEnergy#749] [Q&A] [answered]
Is there a way to ignore or disable old meters so they're not even retrieved by the integration?

I recently joined Octopus and very quickly had smart meters installed, to the point that the old meters only had 1 manual reading associated with them. I've disabled the devices in HA, but as the meters are still listed on my account for historical reasons I still have a whole set of "useless" entities registered. It would be nice to be able to tell the integration to completely ignore meters matching a specific meter serial number and remove any stale entities.

**Comment by BottlecapDave** [comment]
Hello. The integration only ever pulls in meters that have an active tariff against it. If the tariff has changed against the old meters, then a reload of the integration should see the entities ignored. Once done, you'll unfortunately need to manually delete the entities from HA (it's how ha works unfortunately).

If a reload is not working, it might be best to post the diagnostics from the electricity meter device here so we can work out why they're not getting omitted.

**Comment by psfshr** [comment]
Hi.

First, here's my diagnostics...

```
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2024.2.1",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.12.1",
    "docker": true,
    "arch": "aarch64",
    "timezone": "Europe/London",
    "os_name": "Linux",
    "os_version": "6.1.74-haos",
    "supervisor": "2024.01.1",
    "host_os": "Home Assistant OS 11.5",
    "docker_version": "24.0.7",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "hacs": {
      "version": "1.34.0",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "octopus_energy": {
      "version": "10.1.0",
      "requirements": []
    },
    "mastodon_profile_stats": {
      "version": "1.0.10",
      "requirements": []
    }
  },
  "integration_manifest": {
    "domain": "octopus_energy",
    "name": "Octopus Energy",
    "codeowners": [
      "@bottlecapdave"
    ],
    "config_flow": true,
    "dependencies": [
      "repairs",
      "recorder"
    ],
    "documentation": "https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy",
    "homekit": {},
    "iot_class": "cloud_polling",
    "issue_tracker": "https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues",
    "ssdp": [],
    "version": "10.1.0",
    "zeroconf": [],
    "is_built_in": false
  },
  "data": {
    "id": "A-8772418B",
    "octoplus_enrolled": true,
    "electricity_meter_points": [
      {
        "mpan": "**REDACTED**",
        "meters": [
          {
            "serial_number": "**REDACTED**",
            "is_export": false,
            "is_smart_meter": true,
            "device_id": "**REDACTED**",
            "manufacturer": "1063 - Landis and Gyr",
            "model": "00050204",
            "firmware": "38040602"
          }
        ],
        "agreements": [
          {
            "start": "2024-01-20T00:00:00+00:00",
            "end": null,
            "tariff_code": "E-1R-VAR-22-11-01-H",
            "product_code": "VAR-22-11-01"
          }
        ]
      }
    ],
    "gas_meter_points": [
      {
        "mprn": "**REDACTED**",
        "meters": [
          {
            "serial_number": "**REDACTED**",
            "consumption_units": "m\u00b3",
            "is_smart_meter": true,
            "device_id": "**REDACTED**",
            "manufacturer": "1063 - Landis and Gyr",
            "model": "47720101",
            "firmware": "03035524"
          },
          {
            "serial_number": "**REDACTED**",
            "consumption_units": "",
            "is_smart_meter": false,
            "device_id": null,
            "manufacturer": "U6",
            "model": null,
            "firmware": null
          }
        ],
        "agreements": [
          {
            "start": "2024-01-20T00:00:00+00:00",
            "end": null,
            "tariff_code": "G-1R-VAR-22-11-01-H",
            "product_code": "VAR-22-11-01"
          }
        ]
      }
    ]
  }
}
```

The Landis and Gyr meters are the new ones, the other is the old one. I did note this morning that Octopus have now removed the old Electricity meter from my account, but the old Gas meter is still there.

So first try, I attempted to delete the entities for the old Electricity meter, couldn't delete any. Tried restarting the integration, still couldn't delete, had to restart HA before they would mark as "unavailable" and I could delete them. (Also had to go in and re-enable any disabled entities, and restart HA again before I could remove the final entities).

However, the old Electricity meter does still get listed under the Integration (have tried a restart of the Integration and of HA). For the gas meter, can't delete any entities at all (which I assume is expected) as Octopus still have it as "present" on my account, so it still also shows under the Integration:

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/9417766/f8878183-6bb4-4f0f-8336-45629d8398b7)

And the old Electricity and Gas meters detail shows:

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/9417766/3125a934-2036-4a41-aa30-20c40a8af5cb)

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/9417766/aecd4736-3c84-4f58-8921-730f25bf51b3)

Have been told by Octopus (prior to yesterday) that it's "normal" for the old meters to stay on the account for billing purposes, but that doesn't tie in with why they've now removed the old Electricity meter but left the old Gas meter in place :man_shrugging:

Cheers,
Paul

**Reply by BottlecapDave** [reply, marked as answer]
Thanks for that. The integration ignores meters that have been marked as inactive by OE. So I'm assuming for some reason they haven't marked it as inactive (or it might have a "generous" active period). This might be something you need to contact OE about as they might have forgotten to mark it as inactive.

Regarding the empty devices in HA, I'll look at providing the ability to cleaning these up in the future.

**Reply by psfshr** [reply]
Just for reference, OE have now removed both old meters from my account, so I've been able to clean up all the old entities.
